### DISCUSSION TITLE
More granular System Exclusive messages in Mido or dedicated library

### DISCUSSION THREAD

**Discussion by jerekapyaho** [mido/mido#356] [Ideas]
Thanks for the inspiring library for processing MIDI! I've been using it on macOS with Core MIDI.

Currently Mido deals with SysEx messages as the raw data between F0 and F7. I've been developing utilities to extract more out of the bytes: parsing the manufacturer information / universal bytes, and going further down into the manufacturer specific parts for KORG, Yamaha etc. synths.

Would this kind of functionality be natural in Mido, or should I make a separate package out of it? I could make a pull request, or I could start a new library project.

As an example of what I have in mind, you could take a look at my [syxpack library for Rust](https://github.com/coniferprod/syxpack-rs) or the [Swift version](https://github.com/coniferprod/SyxPack). I'll most likely be porting this to Python anyway, so it's just a question of where it would land.

**Comment by rdoursenaud** [comment]
I’d love to have a look at what you did.
I was about to venture in the same direction!

IMHO it would be great to have the generic (part of the official MIDI specification) parsing capability into `mido` .
It would nicely fit the "MIDI Objects" philosophy and Python’s "Batteries included" approach.
Specific parsing by manufacturers could inherit from the generic object and be maintained separately as submodules since most people don’t have equipment from each and every brand for testing ;)

**Comment by jerekapyaho** [comment]
First stab at the problem, don't laugh too hard: https://github.com/coniferprod/midisyx
